lib/pg.provider.js

```diff
@@ -0,0 +1,291 @@
+      }
+      const { Category } = res.rows[0];
+      callback(null, Category);
+    });
+  }
+
+  // Get object from GlobalStorage
+  //   id - globally unique object id
+  //   callback - function(err, obj)
+  get(id, callback) {
+    this.getCategoryById(id, (err, category) => {
+      if (err) {
+        callback(err);
+        return;
+      }
+
+      const objectQuery = `SELECT * FROM ${category} WHERE "Id" = $1`;
+      console.log('objectQuery', objectQuery);
+      this.pool.query(objectQuery, [id], (err, res) => {
+        if (err) {
+          callback(err);
+          return;
+        }
+
+        if (res.rowCount === 0) {
+          callback(new GSError(
+            errorCodes.NOT_FOUND,
+            `No object with Id ${id} available`
+          ));
+          return;
+        }
+
+        callback(null, res.rows[0]);
+      });
+    });
+  }
+
+  // Set object in GlobalStorage
+  //   obj - object to be stored
+  //   callback - function(err)
+  set(obj, callback) {
+    if (!obj.Id) {
+      throw new TypeError('Id is not provided');
+    }
+
+    this.getCategoryById(obj.Id, (err, category) => {
+      if (err) {
+        callback(err);
+        return;
+      }
+
+      const validation = metaschema.validate(category, obj);
+      if (!validation.valid) {
+        callback(new GSError(
+          errorCodes.INVALID_SCHEMA,
+          `Invalid schema provided: ${validation.errors.join('; ')}`
+        ));
+        return;
+      }
+
+      let fields = Object.keys(obj)
+        .filter(key => key !== 'Id');
+      const values = fields.map(key => obj[key]);
+      fields = fields.map(escapeIdentifier);
+      const setQuery = `UPDATE ${category} SET (${fields.join(', ')}) = ` +
+        `(${generateQueryParams(fields.length)})`;
+
+      console.log('setQuery:', setQuery);
+      this.pool.query(setQuery, values, err => {
+        callback(err);
+      });
+    });
+  }
+
+  // Create object in GlobalStorage
+  //   obj - object to be stored
+  //   category - string, category to store the object in
+  //   callback - function(err, id)
+  create(obj, category, callback) {
+    const validation = metaschema.validate(category, obj);
+    if (!validation.valid) {
+      callback(new GSError(
+        errorCodes.INVALID_SCHEMA,
+        `Invalid schema provided: ${validation.errors.join('; ')}`
+      ));
+      return;
+    }
+
+    const categoryType = metaschema.categories.get(category).constructor.name;
+    if (categoryType !== 'Local' && categoryType !== 'Object') {
+      this.takeId((err, id) => {
+        if (err) {
+          callback(err);
+          return;
+        }
+
+        createRecord(id);
+      });
+    } else {
+      createRecord(null);
+    }
+
+    function createRecord(id) {
+      let fields = Object.keys(obj)
+        .filter(key => key !== 'Id');
+      const values = fields.map(key => obj[key]);
+      if (id) {
+        fields.push('Id');
+        values.push(id.toString());
+      }
```

Comment:
I don't quite understand how you want this to work; the `fields` array has to be filtered before this `if` block since it's used when creating the `values` array.